### PR TITLE
[platform.hashing] Add 0.3.0 version

### PR DESCRIPTION
**Pull request by FreePhoenix888**
Specify library name and version:  **platform.hashing/0.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@FreePhoenix888 do not mix tab with space. Please, use only spaces.

**Comment by uilianries** [comment]
@FreePhoenix888 sorry, your current error is internal. I gonna restart the CI job.

File: recipes/platform.hashing/all/conandata.yml [hunk 1]
```diff
@@ -2,3 +2,6 @@ sources:
   "0.2.0":
     url: https://github.com/linksplatform/Hashing/archive/refs/tags/0.2.0.zip
     sha256: 8e34a10d2768d0f2d9ba0cbb88ce193b32245c5a46f14f7ec860c9fddbe8f39f
+  "0.3.0"
```
Comment (SSE4) [inline recipes/platform.hashing/all/conandata.yml:5]:
```suggestion
  "0.3.0":
```

Comment (FreePhoenix888) [reply recipes/platform.hashing/all/conandata.yml:5]: Thank you, I have not seen that error

Comment (madebr) [reply recipes/platform.hashing/all/conandata.yml:5]: This is not fixed yet.

File: recipes/platform.hashing/all/conandata_BACKUP_22189.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+sources:
+  "0.2.0":
+    url: https://github.com/linksplatform/Hashing/archive/refs/tags/0.2.0.zip
+    sha256: 8e34a10d2768d0f2d9ba0cbb88ce193b32245c5a46f14f7ec860c9fddbe8f39f
+<<<<<<< HEAD
```
Comment (SSE4) [inline recipes/platform.hashing/all/conandata_BACKUP_22189.yml:5]: bad merge?

File: recipes/platform.hashing/all/conanfile.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile, CMake, tools
```
Comment (SSE4) [inline recipes/platform.hashing/all/conanfile.py:1]:
```suggestion
from conan import ConanFile
from conan.tools.cmake import CMake
```
and other tools are imported from various namespaces in v2

Comment (FreePhoenix888) [reply recipes/platform.hashing/all/conanfile.py]: @SSE4 Where can I find python conan package docs? [This one](https://pypi.org/project/conan/) brings me to https://docs.conan.io/en/latest/ . I want to know where should I import `tools` from

Comment (FreePhoenix888) [reply recipes/platform.hashing/all/conanfile.py]: @uilianries maybe you can help me? :)

**Comment by FreePhoenix888** [comment]
Where should I import tools from? @uilianries , @SSE4

**Comment by Konard** [comment]
Can we get a little bit of help?

**Comment by SSE4** [comment]
> * platform.hashing/0.2.0@

no problem, I have pushed changes needed

File: recipes/platform.hashing/all/conanfile.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.cmake import CMake
+from conan.errors import ConanInvalidConfiguration
```
Comment (SSE4) [inline recipes/platform.hashing/all/conanfile.py:3]:
```suggestion
+try:
+    from conan.tools.build import check_min_cppstd
+except ImportError:
+    from conans.tools import check_min_cppstd  # FIXME : not in 1.49
+from conan.tools.files import get
 from conan.tools.cmake import CMake
+from conan.tools.scm import Version
```

**Comment by SSE4** [comment]
sorry, can't push (`allow edits from maintaintainers` wasn't checked)
here is the patch:
```
diff --git a/recipes/platform.hashing/all/conanfile.py b/recipes/platform.hashing/all/conanfile.py
index 9e3d7d51b..c0f03a178 100644
--- a/recipes/platform.hashing/all/conanfile.py
+++ b/recipes/platform.hashing/all/conanfile.py
@@ -1,5 +1,11 @@
 from conan import ConanFile
+try:
+    from conan.tools.build import check_min_cppstd
+except ImportError:
+    from conans.tools import check_min_cppstd  # FIXME : not in 1.49
+from conan.tools.files import get
 from conan.tools.cmake import CMake
+from conan.tools.scm import Version
 from conan.errors import ConanInvalidConfiguration
 import os
 
@@ -48,14 +54,14 @@ class PlatformInterfacesConan(ConanFile):
             self.output.warn("{} recipe lacks information about the {} compiler support.".format(
                 self.name, self.settings.compiler))
 
-        elif tools.Version(self.settings.compiler.version) < minimum_version:
+        elif Version(self.settings.compiler.version) < minimum_version:
             raise ConanInvalidConfiguration("{}/{} requires c++{}, "
                                             "which is not supported by {} {}.".format(
                 self.name, self.version, self._minimum_cpp_standard, self.settings.compiler,
                 self.settings.compiler.version))
 
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+            check_min_cppstd(self, self._minimum_cpp_standard)
 
         if self.settings.arch in ("x86", ):
             raise ConanInvalidConfiguration("{} does not support arch={}".format(self.name, self.settings.arch))
@@ -64,8 +70,7 @@ class PlatformInterfacesConan(ConanFile):
         self.info.header_only()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
 
     def package(self):
         self.copy("*.h", dst="include", src=self._internal_cpp_subfolder)
@@ -82,6 +87,6 @@ class PlatformInterfacesConan(ConanFile):
             }.get(str(self.settings.arch), "")
         self.user_info.suggested_flags = suggested_flags
 
-        if "-march" not in "{} {}".format(tools.get_env("CPPFLAGS", ""), tools.get_env("CXXFLAGS", "")):
+        if "-march" not in "{} {}".format(os.environ.get("CPPFLAGS", ""), os.environ.get("CXXFLAGS", "")):
             self.output.warn("platform.hashing needs to have `-march=ARCH` added to CPPFLAGS/CXXFLAGS. "
                              "A suggestion is available in deps_user_info[{name}].suggested_flags.".format(name=self.name))
-- 
2.33.0.windows.2


```

**Review by SSE4 — APPROVED** [review]

**Pull request merged**